service/src/main/java/org/dice_research/fc/sparql/query/QueryEngineCustomHTTP.java

```diff
@@ -0,0 +1,225 @@
+        this.query = query;
+        this.service = service;
+    }
+
+    @Override
+    public void setInitialBinding(QuerySolution binding) {
+
+    }
+
+    @Override
+    public Dataset getDataset() {
+        return null;
+    }
+
+    @Override
+    public Context getContext() {
+        return null;
+    }
+
+    @Override
+    public Query getQuery() {
+        return this.query;
+    }
+
+    @Override
+    public ResultSet execSelect() {
+        String result = createRequest(query.toString(), timeout, service);
+        ResultSetFactory fac = new ResultSetFactory();
+
+        // the result is not a valid XML then replace with an empty XML
+        if(result.length()<10) {
+            result = emptyXML();
+        }
+
+        ResultSet resultSet = fac.fromXML(result);
+
+        return resultSet;
+    }
+
+    private String emptyXML() {
+        return "<sparql xmlns=\"http://www.w3.org/2005/sparql-results#\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.w3.org/2001/sw/DataAccess/rf1/result2.xsd\"><head></head><results distinct=\"false\" ordered=\"true\"></results></sparql>>";
+    }
+
+    private static String read(InputStream content) {
+        ByteSource byteSource = new ByteSource() {
+            @Override
+            public InputStream openStream() throws IOException {
+                return content;
+            }
+        };
+
+        try {
+            return byteSource.asCharSource(Charsets.UTF_8).read();
+        } catch (IOException e) {
+            LOGGER.error("Could not read stream due to ",e);
+        }
+        return "";
+
+    }
+
+    protected static String createRequest(String sparqlQuery, Long to,String service){
+        List<String> ret =  new ArrayList<String>();
+        int code=0;
+        String test;
+        String actualContentType="";
+        try {
+            int timeout = to.intValue();
+            RequestConfig config = RequestConfig.custom()
+                    .setConnectTimeout(timeout)
+                    .setConnectionRequestTimeout(timeout)
+                    .setSocketTimeout(timeout).build();
+            CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+            HttpGet get = new HttpGet(service+"?query="+ URLEncoder.encode(sparqlQuery, "UTF-8"));
+            get.addHeader(HttpHeaders.ACCEPT, "application/sparql-results+xml");
+            HttpResponse resp = client.execute(get);
+            test = read(resp.getEntity().getContent());
+            InputStream is = new ByteArrayInputStream(test.getBytes(StandardCharsets.UTF_8));
+            String result = IOUtils.toString(is, StandardCharsets.UTF_8);
+            LOGGER.info(result);
+            return result;
+        }catch(Exception e){
+            LOGGER.error("Could not execute request due to ",e);
+        }
+        return "";
+    }
+
+    @Override
+    public Model execConstruct() {
+        return null;
```

Comment:
Please 1) document in the class description that these methods are not implementation and 2) use something like `throw new NotImplementedException("The class " + this.getClass().toString() + " does not implement this method.");`

You can also use the `UnsupportedOperationException` that you used further below :wink: